x-pack/plugins/ml/public/application/jobs/new_job/pages/components/common/datafeed_preview_flyout/datafeed_preview.tsx

```diff
@@ -63,18 +63,9 @@ export const DatafeedPreview: FC<{
 
     if (combinedJob.datafeed_config && combinedJob.datafeed_config.indices.length) {
       try {
-        const resp = await mlJobService.searchPreview(combinedJob);
-        let data = resp.hits.hits;
-        // the first item under aggregations can be any name
-        if (isPopulatedObject(resp.aggregations)) {
-          const accessor = Object.keys(resp.aggregations)[0];
-          const aggregate = resp.aggregations[accessor];
-          if (isMultiBucketAggregate(aggregate)) {
-            data = aggregate.buckets.slice(0, ML_DATA_PREVIEW_COUNT);
-          }
-        }
-
-        setPreviewJsonString(JSON.stringify(data, null, 2));
+        const { datafeed_config: datafeed, ...job } = combinedJob;
+        const preview = await datafeedPreview(undefined, job, datafeed);
```

Comment:
Before adding a detector, the flyout now shows this. Maybe it would be better not to run the preview before adding a detector? The previous approach showed docs with timestamps.

![image](https://user-images.githubusercontent.com/7405507/121520354-bb7d1880-c9ea-11eb-96e4-1caa17d4ce46.png)

Reply:
I've added a check for the detectors list. if it is empty the preview isn't run and and a message is shown.

![image](https://user-images.githubusercontent.com/22172091/121542686-40bef800-ca00-11eb-9bf3-797bacf38723.png)